greengrass_defender_agent/agent.py

```diff
@@ -106,13 +108,26 @@ def publish_metrics(ipc_client, config_changed, metrics_collector, sample_interv
 def main():
     # Get the ipc client
     ipc_client = ipc_utils.IPCUtils()
```

Comment:
Not sure which python Device SDK version you're testing with. But now it has a v2 GG IPC client in newer SDK now and it's preferred and easier to use: https://docs.aws.amazon.com/greengrass/v2/developerguide/interprocess-communication.html#ipc-service-connect maybe going forward with this component your team would want to use the new one

Reply:
I am aware of this outdated dependency problem. But during the troubleshooting phase I decided to not introduce another factor of change. Was planning to create a follow up action item after this problem is closed.